agent-tooling/src/main/java/io/opentelemetry/auto/tooling/HelperInjector.java

```diff
@@ -142,12 +142,14 @@ public static HelperInjector forDynamicTypes(
             helperModules.add(new WeakReference<>(javaModule.unwrap()));
           }
         } catch (final Exception e) {
-          log.error(
-              "Error preparing helpers while processing {} for {}. Failed to inject helper classes into instance {}",
-              typeDescription,
-              requestingName,
-              classLoader,
-              e);
+          if (log.isErrorEnabled()) {
```

Comment:
I don’t think we need these `ifs` polluting the code

Reply:
Did you see https://github.com/DataDog/dd-trace-java/pull/1466? I'm not opposed to it in this case for avoiding memory allocation.